### DISCUSSION TITLE
Fuel redesign

### DISCUSSION THREAD

**Discussion by royfalk** [vegastrike/Vega-Strike-Engine-Source#885] [Ideas]
Current
-----------
So the current fuel-drive is cryptic to say the least. 

```
float Energetic::ExpendMomentaryFuelUsage(float magnitude) {
    // TODO: have this make some kind of sense to someone other than the person who wrote the comment below.
    //HACK this forces the reaction to be Li-6+D fusion with efficiency governed by the getFuelUsage function
    float quantity = Energetic::getFuelUsage(false) * simulation_atom_var * magnitude *
            configuration()->fuel.fmec_exit_velocity_inverse / configuration()->fuel.fuel_efficiency;

    return ExpendFuel(quantity);
}
```

Suggestion
----------------
I suggest refactoring it as follows:

FT - flight time
FT/sec - flight time/second

1. Numbers will be relative to ship size, not absolute. 
2. Fuel number will denote minutes of flight time, not metric tons.
3. Drive will consume 1 FT/sec.
4. Afterburner will consume 2 FT/sec.
5. AfterburnerUpgrade and DriveUpgrade will modify above.
6. Reactor01 will consume 1 FT/sec.
7. Larger reactors will consume accordingly.
8. Calculation will take 15 from config.

In practice
---------------
In practice this will mean the following:
Say we want a Gawain (reactor04) to have 10 minutes of actual flight. 
reactor04 = 57 -> 3.8 FT/sec.
10 minutes at 1 FT/sec.
90 seconds of afterburner at 2 FT/sec.

Add it all up and you get 51 minutes. It's an easy number to grasp intuitively.
50/(1 + 4 + ?) ~= 10 minutes

**Pros**
1. Remove several unnecessary fields from both ship files and configuration. fmec_exit_velocity_inverse, fuel.fuel_efficiency, afterburner cost, etc. Move others from ship to configuration. 
2. Replace sometimes meaningless fuel field - Agasicles carrying 1,819,671 metric tons of fuel. Ancestor carries 3.25 for a 100+ ton ship.  Replace it with minutes of flight/game time.

** Cons **
1. Description is eh. Metric tons sounds cooler.  Maybe fudge some formula in description?
2. Refuel cost also needs to factor in ship mass.

**Comment by evertvorster** [comment]
Talking about fuel, I noticed in the most recent version of the game that my fuel goes down very, very fast when flying around. I double-checked that the ship had a little repair and refuel, and still I was stunned by the rate at which it was dropping... about 25% fuel used in just a couple of minutes of flying. Something changed here, and not for the better.. 
I'll record some video to show what I am talking about, but this would probably need its own issue to be raised. 

I'd say that ten minutes of flight is very, very restrictive, I would expect many hours of flying on a full tank, just like with a car?

**Comment by royfalk** [comment]
> I'd say that ten minutes of flight is very, very restrictive, I would expect many hours of flying on a full tank, just like with a car?

First off, there's game time (GT) and real time (RT). Real time 10 minutes translates to several days of game time. 

More to the point:
1. We want fuel to be an issue. If you need to refuel every 90 minutes (RT), you may as well not have it at all. 
2. We want fighters to be more limited but fast. We want upgrades to fighters to be offset by something and fuel consumption is a great way.
3. I assume you were flying the Llama. It has a lot of fuel and a lot of range.
4. The current ships have inconsistent fuel numbers. I don't know what they were thinking when they chose these numbers. Literally. There's no design doc.
5. The new proposal makes it very clear what we want from each ship. 30 minutes fuel and it's a very short range fighter defending an installation or carrier. 6 hours - capital ship or merchant. Lots of range.
6. I mostly deal with the engine not the data. If something needs to change to fit, I prefer to change the data to fit the engine and not the other way around.
7. Because the old stuff is so inconsistent, I don't care that I'm breaking it, as long as I (or you) fix it. 
8. We can have a configuration option for limitless fuel for people who prefer not to deal with this. 

Please don't waste your time on a video. I believe you.
